tests/testDipoleFitter.py

```diff
@@ -0,0 +1,252 @@
+from __future__ import absolute_import, division, print_function
+#
+# LSST Data Management System
+# Copyright 2008-2016 AURA/LSST.
+#
+# This product includes software developed by the
+# LSST Project (http://www.lsst.org/).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <https://www.lsstcorp.org/LegalNotices/>.
+
+import unittest
+
+import numpy as np
+
+import lsst.utils.tests
+import lsst.afw.table as afwTable
+import lsst.meas.base as measBase
+from lsst.ip.diffim.dipoleFitTask import (DipoleFitAlgorithm, DipoleFitTask)
+import lsst.ip.diffim.utils as ipUtils
+
+# This file produces three unit tests of the DipoelFitAlgorithm and its related tasks and plugins.
+# Each test generates a fake image with two synthetic dipoles as input data.
+#
+# DipoleFitTest.testDipoleAlgorithm - tests the DipoleFitAlgorithm directly and ensures that the recovered
+#   fluxes and centroids are close to the input values.
+# DipoleFitTest.testDipoleTask - tests the DipoleFitTask on identical synthetic data, after running
+#   dipole detection on the image.
+# DipoleFitTest.testDipoleEdge - ensures correct handling of dipoles too close to the edge of the image.
+
+
+class DipoleFitTestGlobalParams(object):
```

Comment:
I'm confused why this class isn't a parent of DipoleFitTest, with __init__() -> setup(). Then you get all this stuff for "free" with each test, and all your params.stuff become self.stuff.